composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -64,25 +67,32 @@ def set_batch_sequence_length(batch: Dict[str, torch.Tensor], curr_seq_len: int,
     """
 
     assert isinstance(batch, Mapping)
+    if 'attention_mask' not in batch:
```

Comment:
only if `preserve_end_of_sequence` is True though, right?

Reply:
Yes, good point. We can relax this requirement to that case. Will do.